pilot/pkg/xds/endpoint_builder.go

```diff
@@ -425,10 +460,54 @@ func buildEnvoyLbEndpoint(proxyless bool, e *model.IstioEndpoint) *endpoint.LbEn
 			},
 		}
 	}
+	if dir == model.TrafficDirectionInboundVIP {
```

Comment:
Is it possible to move these to a waypoint endpoint builder like listener and cluster?

Reply:
It may be a bit hard since this buildEnvoyLbEndpoint is a common function used by all types, we just have a small bit of custom logic here